pkg/webhook/pod/mutating/pod_create_update_handler.go

```diff
@@ -46,31 +46,49 @@ var _ admission.Handler = &PodCreateHandler{}
 // Handle handles admission requests.
 func (h *PodCreateHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
 	obj := &corev1.Pod{}
-
-	err := h.Decoder.Decode(req, obj)
-	if err != nil {
-		return admission.Errored(http.StatusBadRequest, err)
-	}
-	var copy runtime.Object = obj.DeepCopy()
-	// when pod.namespace is empty, using req.namespace
-	if obj.Namespace == "" {
-		obj.Namespace = req.Namespace
-	}
-
-	injectPodReadinessGate(req, obj)
-
-	err = h.sidecarsetMutatingPod(ctx, req, obj)
```

Comment:
sidecarsetMutatingPod contains Pod update, suggest function Handle should not understand specific pod mutating logic (e.g., update, create, delete).